xmtp_mls/src/client.rs

```diff
@@ -465,6 +467,7 @@ where
         Ok(group)
     }
 
+    #[cfg(feature = "message-history")]
```

Comment:
not sure about this one. AFAIK, sync groups can/will be used for other purposes.

Reply:
Fine to gate for now, and if it breaks something we can always un-gate.